src/analysis/functions.rs

```diff
@@ -36,14 +37,14 @@ use std::{
 use super::special_functions;
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum Visibility {
+pub enum FuncVisibility {
```

Comment:
I'm not really sure how this enum type is fit for the rest of the code. It confuses two things, a function visibility and whether gir is capable of generating the code or not (when it's set to Comment). I suppose we should ideally drop it and move the Comment to a new Status mode for "not capable of generating"

Reply:
I have dropped the enum and replaced it with two variants of GStatus, ideally the later (GStatus::Hidden) should be a different attribute of the analysis::Info, any opinions?

Reply:
with this change, it gets pretty interesting; we can finally drop all the ignore=true we had in Gir.toml in order to not have a doc link because the function can't be generated for whatever reasons